File: plugin/zeppelin/src/main/scala/org/intellij/plugin/zeppelin/api/idea/IdeaEditorApi.scala
package org.intellij.plugin.zeppelin.api.idea

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.editor.Editor
import com.intellij.openapi.fileEditor.ex.FileEditorManagerEx

/**
  * API for working with current Editor
  */
trait IdeaEditorApi {
  def currentEditor(anActionEvent: AnActionEvent): Option[Editor] = {
    Option(FileEditorManagerEx.getInstanceEx(anActionEvent.getProject).getSelectedTextEditor)
  }

  def currentSelectedText(editor: Option[Editor]): String = {
    if (editor.isEmpty) return ""
    val selectionModel = editor.head.getSelectionModel
    val blockStarts = selectionModel.getBlockSelectionStarts
    val blockEnds = selectionModel.getBlockSelectionEnds
    editor.head.getDocument.getCharsSequence.subSequence(blockStarts(0), blockEnds(0)).toString
  }
}

File: zeppelin/implementation/src/org/intellij/plugin/zeppelin/api/idea/IdeaEditorApi.kt
package org.intellij.plugin.zeppelin.api.idea

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.editor.Editor
import com.intellij.openapi.editor.SelectionModel
import com.intellij.openapi.fileEditor.ex.FileEditorManagerEx

/**
 * API for working with current Editor
 */
object IdeaEditorApi {
    fun currentEditor(anActionEvent: AnActionEvent): Editor? {
        return FileEditorManagerEx.getInstanceEx(anActionEvent.project ?: return null).selectedTextEditor
    }

    fun currentSelectedText(editor: Editor?): String {
        val selectionModel: SelectionModel = editor?.selectionModel ?: return ""
        val blockStarts: IntArray = selectionModel.blockSelectionStarts
        val blockEnds: IntArray = selectionModel.blockSelectionEnds
        return editor.document.charsSequence.subSequence(blockStarts[0], blockEnds[0]).toString()
    }
}